### ISSUE TITLE
Links are ugly and filled with unnecessary symbols

### ISSUE THREAD

**Issue by ben-pace** [LessWrong2/Lesswrong2#922]
I don't think we need JiMAMNAb55Qq24nES <- that many slots in our links to identify posts by. Can we please have like 4 slots until we need more?

**Comment by jimrandomh** [comment]
The way this worked on LW1 was, IDs were allocated sequentially and the links contained the ID in base36. Whereas in Vulcan, link IDs are allocated randomly and apparently are 17 digits of base62. To get nice short links, we would:
  1. Rename `legacyId` to `shortId`
  2. Start allocating `shortId`s to new posts/comments, and work out the concurrency issues that entails
  3. Allocate a `shortId` to existing posts/comments
  4. Change URL routing and links to use `shortId`s

**Comment by Discordius** [comment]
Yep, I think this is relatively straightforward, the problem is just that
we would add a layer of redirect to all links that were made in the
meantime, which does come with some loading time costs.
